engine/lib/views.php

```diff
@@ -1342,20 +1373,19 @@ function elgg_view_tagcloud(array $options = array()) {
  *
  * @param \ElggEntity|\ElggAnnotation $item
  * @param array  $vars Additional parameters for the rendering
- *
+ *      'entity_view'     Alternative view to render ElggEntity
+ *      'annotation_view' Alternative view to render ElggAnnotation
+ *      'river_view'      Alternative view to render ElggRiverItem
  * @return string
  * @since 1.8.0
  * @access private
  */
 function elgg_view_list_item($item, array $vars = array()) {
-	global $CONFIG;
-
-	$type = $item->getType();
-	if (in_array($type, $CONFIG->entity_types)) {
```

Comment:
I know I suggested refactoring an elgg_view_exists() call before, I was wrong. Putting unrelated refactoring into this PR will make it harder to evaluate. This is good but could go in another PR.